src/main/java/edu/cornell/kfs/tax/batch/TransactionOverrideCsvBatchInputFileType.java

```diff
@@ -0,0 +1,443 @@
+        return transactionOverrides;
+    }
+
+    /**
+     * Transforms each parsed line into a TransactionOverride BO if content is valid,
+     * or into a String representation of the parsed line if content is invalid.
+     * Assumes that TransactionOverrideCsv is the header enum.
+     * 
+     * @param parsedLine The line to process, as a Map from header names to values.
+     * @param lineNumber The current line number, starting from 1.
+     * @param docNumberMaxLength The max length of the BO's document number property.
+     * @param boxNumberMaxLength The max length of the BO's tax box property.
+     * @return A TransactionOverride object if a valid line, otherwise a tab-delimited String representation of the line.
+     */
+    protected Object convertParsedLineToVO(Map<String,String> parsedLine, int lineNumber, int docNumberMaxLength, int boxNumberMaxLength) {
+        boolean valid = true;
+        TransactionOverride transOverride = new TransactionOverride();
+        
+        // Verify that exactly one type of tax box was specified, and that the tax box has the expected max length. Also setup tax box properties if valid.
+        String box1099 = parsedLine.get(TransactionOverrideCsv.Form_1099_Box.toString());
+        String box1042S = parsedLine.get(TransactionOverrideCsv.Form_1042S_Box.toString());
+        if (StringUtils.isBlank(box1099)) {
+            if (StringUtils.isBlank(box1042S)) {
+                LOG.error("Found a line that does not specify a 1099 or 1042S tax box override. Line number: " + Integer.toString(lineNumber));
+                valid = false;
+            } else if (box1042S.length() > boxNumberMaxLength) {
+                LOG.error("Found a line with a 1042S box number that is too long. Line number: " + Integer.toString(lineNumber));
+                valid = false;
+            } else {
+                // Validation succeeded; configure override of a 1042S tax box.
+                transOverride.setTaxType(CUTaxConstants.TAX_TYPE_1042S);
+                transOverride.setBoxNumber(NULL_STRING.equalsIgnoreCase(box1042S) ? CUTaxConstants.TAX_1042S_UNKNOWN_BOX_KEY : box1042S);
+            }
+        } else if (StringUtils.isNotBlank(box1042S)) {
+            LOG.error("Found a line that specifies both a 1099 and 1042S tax box override. Line number: " + Integer.toString(lineNumber));
+            valid = false;
+        } else if (box1099.length() > CUTaxConstants.TAX_1099_MAX_BUCKET_LENGTH && !NULL_STRING.equalsIgnoreCase(box1099)) {
+            LOG.error("Found a line with a 1099 box number that is too long. Line number: " + Integer.toString(lineNumber));
+            valid = false;
+        } else {
+            // Validation succeeded; configure override of a 1099 tax box.
+            transOverride.setTaxType(CUTaxConstants.TAX_TYPE_1099);
+            transOverride.setBoxNumber(NULL_STRING.equalsIgnoreCase(box1099) ? CUTaxConstants.TAX_1099_UNKNOWN_BOX_KEY : box1099);
+        }
+        

```

Comment:
The TransactionOverride objects do have a tax type field to indicate which type of tax processing the override is for. However, the current transaction detail table (which the transaction detail output file sources from) uses separate columns for 1099 and 1042S box numbers, so that's reflected in the output file. I went with preserving the column separation in the input file so that, if the tax managers just wanted to modify the output file to create the input one, they wouldn't have to add a new column.

There are several other places in the code that rely on the transaction detail table having separate 1099 and 1042S box number columns, so I do not want to change those at this time. If you think that the raw transaction detail output file should consolidate those columns and add a new tax-type indicator column instead, let me know and I can check if the tax managers are okay with that change.

I would prefer not to add a new parameter for controlling what tax type the uploaded overrides are for, since I think that would be overkill.